### DISCUSSION TITLE
v4 API - filtering by date

### DISCUSSION THREAD

**Discussion by jake-anylaw** [freelawproject/courtlistener#4709] [Q&A] [answered]
Hi-- I'm a new dev for anylaw. Trying to backfill some specific years and jurisdictions. I'm trying to use the filters, specifically date_created_lte. For example, if I make a GET request like this: "https://www.courtlistener.com/api/rest/v4/dockets?court=nysupct&date_created__lte=2014-10-31&order_by=date_created" - I'd expect results from before that date, but I get null. If change to date_created__lte=2016-10-31, it returns results. If I check via the Courtlistener search page, there are results from before 2014-10-31. Is there some sort of cutoff with the filter or am I not understanding this correctly? Appreciate it!!

**Comment by mlissner** [comment, marked as answer]
I think you've got it, but what's a bit confusing is that the search engine has a trick that it does behind the scenes when you search for `nysupct`. We have a system of parent-child courts that allows us to tag cases with the greatest-possible granularity. So when you search for `nysupct` on the search engine, it expands your query to also include the child courts for `nysupct`, but when you're on the API, we don't do that. The idea is that search is fuzzy and can do nice things for users, but the API is more like a database and should do exactly what you tell it to.

The fix for this, I think, is to find the child courts for `nysupct`, and then filter by those in the API as well.

**Reply by jake-anylaw** [reply]
Thank you for the quick reply. So you're thinking, since I'm missing the child courts and there may not be any cases within my date range in the main parent court--  that's why I get no results? I just wanted to make sure I wasn't missing some sort of limit on the date filter. I didn't see anything when I looked through filter options. Will test with added child courts and also on some other jurisdictions and see how it goes. Thanks for putting this all together!

**Reply by mlissner** [reply]
Yep, I think that's exactly right!

**Reply by jake-anylaw** [reply]
Awesome, thanks-- will report back but I think this is likely resolved!
